lib/network/api/api.go

```diff
@@ -31,16 +33,24 @@ type NetworkHandlerAPI struct {
 	localNode *node.LocalNode
 	network   network.Network
 	storage   *storage.LevelDBBackend
+	urlPrefix string
+	version   string
 }
 
-func NewNetworkHandlerAPI(localNode *node.LocalNode, network network.Network, storage *storage.LevelDBBackend) *NetworkHandlerAPI {
+func NewNetworkHandlerAPI(localNode *node.LocalNode, network network.Network, storage *storage.LevelDBBackend, urlPrefix string) *NetworkHandlerAPI {
 	return &NetworkHandlerAPI{
 		localNode: localNode,
 		network:   network,
 		storage:   storage,
+		urlPrefix: urlPrefix,
+		version:   APIVersionV1,
 	}
 }
 
+func (api NetworkHandlerAPI) HandlerURLPattern(pattern string) string {
+	return fmt.Sprintf("%s/%s%s", api.urlPrefix, api.version, pattern)
```

Comment:
It feels weird to have `%s/%s%s` and not `%s/%s/%s` but I guess `pattern` should always start with `/` ?
In any case, I think we should use `url.Url` here instead of manual building via `Sprintf`.

Reply:
@Geod24  As you know, UrlPrefix has `/` too. and `url.Url.Path` is just string. Do you think `path.Join` instead `Sprintf` or string+string?

Reply:
```
type URL struct {
        Scheme     string
        Opaque     string    // encoded opaque data
        User       *Userinfo // username and password information
        Host       string    // host or host:port
        Path       string    // path (relative paths may omit leading slash)
        RawPath    string    // encoded path hint (see EscapedPath method)
        ForceQuery bool      // append a query ('?') even if RawQuery is empty
        RawQuery   string    // encoded query values, without '?'
        Fragment   string    // fragment for references, without '#'
}
```
This pattern is just  url.Path.

Reply:
@Geod24 If this approach (using `HandlerURLPattern`) is not bad, I want to merge it. Because I have to finish this iteration ㅠㅠ  I will change inside this func with `path.Join`.